### PR TITLE
Clarity-Bitcoin V5

### PR DESCRIPTION
**Pull request by MarvinJanssen**
Current additions:
- Segwit TXID calculation. It was missing in v4. Since the TXID calculation requires some work, it needs to be requested by the caller via the new boolean parameter. 
- `parse-wtx` now returns `ERR-NOT-SEGWIT-TRANSACTION` (`(err u12)`) if provided with a non-segwit transaction. In v4 it returns an empty transaction tuple.
- `parse-tx` and `parse-wtx` no longer allow leftover data. In v4 one could append arbitrary data at the end of the Bitcoin transaction and the library would parse the transaction normally but it would use the leftover data for TXID calculation, leading to the wrong TXID and potential security risks. The functions are now more strict and will return `ERR-LEFTOVER-DATA` (`(err u13)`) if there is any leftover data.
- Non-compact version `was-tx-mined` is removed, does not appear to be used and is inefficient. It also had no wtx counterpart. See #26.

We can do optimisations once we the feature set of V5 is fully scoped.

### CODE REVIEW

**Comment by friedger** [comment]
LGTM

**Comment by friedger** [comment]
@MarvinJanssen The PR description is good as changelog, we have a change log for deployed version in README.md that will be updated once deployed. 

Any more ideas for improvements?

File: contracts/clarity-bitcoin.clar [hunk 1]
```diff
@@ -1,7 +1,5 @@
 ;; @contract stateless contract to verify bitcoin transaction
-;; @version 4
-
-;; version 4 adds support for segwit transactions
+;; @version 5
```
Comment (friedger) [inline contracts/clarity-bitcoin.clar:2]: A one liner about the changes in v5 should go here as well.

**Comment by MarvinJanssen** [comment]
Are we going to scope v5 to just the changes made in this PR? I was told by @setzeus that people are waiting for v5 of the library and need to start using it. In that case, are there any feature requests we should still bring into it now?

**Comment by setzeus** [comment]
@MarvinJanssen just the changes in PR for now works* we can push a 5.x version later if any feature is requested.

**Review by friedger — APPROVED** [review]
LGTM

**Pull request merged**